### PR TITLE
[IMPROVE] Add `file-title` and `file-desc` as new filter tag options on message search

### LINKED ISSUE #11584: [IMPROVEMENT] Take attachment comments into account when message searching
Currently, when you upload a file you can't find it by searching messages.

### PR DESCRIPTION
**Pull request by subham103**
closed #11584 
Currently, when we upload a file we can't find it by searching messages. Add option for search in a message using file title (title:TITLE_NAME) and file description (desc:DESCRIPTION)

pr7_1

### CODE REVIEW

File: server/methods/messageSearch.js [hunk 1]
```diff
@@ -117,6 +117,16 @@ Meteor.methods({
 			return '';
 		}
 
+		function filterTitle(_, tag) {
+			query['attachments.0.title'] = new RegExp(s.escapeRegExp(tag), 'i');
```
Comment (rodrigok) [inline server/methods/messageSearch.js:121]:
```suggestion
			query['attachments.title'] = new RegExp(s.escapeRegExp(tag), 'i');
```

File: server/methods/messageSearch.js [hunk 2]
```diff
@@ -171,6 +181,10 @@ Meteor.methods({
 		text = text.replace(/has:location|has:map/g, filterLocation);
 		// Filter image tags
 		text = text.replace(/label:(\w+)/g, filterLabel);
+		// Filter on description of messages.
+		text = text.replace(/desc:(\w+)/g, filterDescription);
```
Comment (rodrigok) [inline server/methods/messageSearch.js:185]:
```suggestion
		text = text.replace(/file-desc:(\w+)/g, filterDescription);
```

File: server/methods/messageSearch.js [hunk 3]
```diff
@@ -171,6 +181,10 @@ Meteor.methods({
...
+		// Filter on title of messages.
+		text = text.replace(/title:(\w+)/g, filterTitle);
```
Comment (rodrigok) [inline server/methods/messageSearch.js:187]:
```suggestion
		text = text.replace(/file-title:(\w+)/g, filterTitle);
```

**Comment by luixxiul** [comment]
Will this option be enabled by default on `3.2.0`?

**Comment by rodrigok** [comment]
@luixxiul yes

**Comment by luixxiul** [comment]
Thanks for the quick reply 😄

**Pull request merged**